### ISSUE TITLE
TensorFlow Android 1.2 - Adds READ_PHONE_STATE permission

### ISSUE THREAD

**Issue by meavydev** [tensorflow/tensorflow#12460]

**Comment by meavydev** [comment]
I am building with:
     compile 'org.tensorflow:tensorflow-android:1.2.0'
My app was rejected by Play Store due to having the READ_PHONE_STATE permission and no privacy policy. However I don't have this permission anywhere in my app, and tracked it down to TensorFlow adding it.
Judging by this thread, I am not alone:
https://stackoverflow.com/questions/44864081/tensorflow-android-library-asking-for-read-phone-state

**Comment by andrewharp** [comment]
@meavydev If you add targetSdkVersion = 23 to your build.gradle and/or minSdkVersion = 19, does this solve the issue? READ_PHONE_STATE is automatically requested by the application target/minSdkVersion are < 4. We set this in AndroidManifest.xml, but only compileSdkVersion in the build.gradle, in case you've copied that for your app.

If you only get this issue when including the TF AAR, then we should probably figure out how to add it to the AAR build as well.

**Comment by meavydev** [comment]
I am only using the AAR, not building from source.
I am building with target SDK set to 24 and build tools on 24, due to a tool that only works with that setup.
My minSDK is set to 16.
Given it sounds like it's an AAR issue, I can try a build increasing the minSDK, but it was running at this level for some AR glasses.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 277 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 292 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 307 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by PaulWoitaschek** [comment]
I guess this just nees a min-sdk like #20454

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by crobertsbmw** [comment]
@PaulWoitaschek I'm having this issue. And users are wondering why I need this permission (I don't). Any update on this?

**Comment by PaulWoitaschek** [comment]
My PR was merged and the next update should fix this.

**Comment by PaulWoitaschek** [comment]
This should be fixed by b148d228886e53f71401c332b8d48d45467dad47

**Comment by crobertsbmw** [comment]
Thanks for the update. The workaround for now is to add this to your manifest:
```
<uses-permission
        android:name="android.permission.READ_PHONE_STATE"
        tools:node="remove" />
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rustyhamsterr** [comment]
While checking why my app required READ_PHONE_STATE I found out some things I want to share:
- crobertsbmw is absolutely right, the workaround is very helpful!
- the fix mentioned by PaulWoitaschek was merged before release of 1.10.0 lib, which still does NOT contain the fix and asks for the permission in question. (I assume a merge readded the code snippet)
- the actual master commit [bf23e5f](https://github.com/tensorflow/tensorflow/tree/bf23e5f0f6e75a35521b7765da9863dbc24d8a71) does again contain the fix and does not require the READ_PHONE_STATE permission

I hope the permission is fixed by next release of tensorflow-lite lib 1.11.0

**Comment by PaulWoitaschek** [comment]
I don't understand what happened here:

**Comment by PaulWoitaschek** [comment]
I don't understand what happened here:
https://github.com/tensorflow/tensorflow/commits/master/tensorflow/contrib/lite/java/AndroidManifest.xml

602b03d5fb521aef6561a5c131075d915907357e My commit from Jul 16, 2018
b148d228886e53f71401c332b8d48d45467dad47 jdduke authored and tensorflower-gardener committed 23 days ago

Two consecutive commits added the same thing to the same file? What am I missing?

**Comment by rustyhamsterr** [comment]
That was a thing I ran into as well.
There are different folders with different manifests. Your commit downgraded the targetSdk for the Android TFLite sample app, as the jdduke commit added the sdk configuration to `tensorflow/contrib/android/cmake/src/main/AndroidManifest.xml`, which I assume to be the source for building the library.

The library manifest got fixed once, but lost the sdk configuration in a commit, somewhere before building version 1.10.0.

Somebody then noticed the mistake, as it is fixed for the mentioned master commit.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by PaulWoitaschek** [comment]
1.11.0 is released but not to jcenter. Can you please push the update to jcenter too?

**Comment by wt-huang** [comment]
@PaulWoitaschek can you please check whether the update has been pushed to jcenter?

**Comment by PaulWoitaschek** [comment]
https://bintray.com/google/tensorflow/tensorflow-lite

Only 1.10.0

**Comment by PaulWoitaschek** [comment]
@wt-huang pingpong

**Comment by wt-huang** [comment]
@PaulWoitaschek thanks for the confirmation. Any ETA as to when the updated 1.11.0 will be pushed to jcenter?

**Comment by PaulWoitaschek** [comment]
I am not involved into that, I just contributed the fix.

**Comment by wt-huang** [comment]
@PaulWoitaschek @meavydev Closing this issue for now as the fix is in for 1.10.0 and 1.11.0 will be pushed to jcenter when the release is ready. Feel free to open a new issue if any error comes up.

**Comment by PaulWoitaschek** [comment]
But 1.10.1?
